homeassistant/components/adguard/sensor.py

```diff
@@ -26,7 +25,7 @@
 class AdGuardHomeEntityDescription(SensorEntityDescription):
     """Describes AdGuard Home sensor entity."""
 
-    value_fn: Callable[[AdGuardHome], Coroutine[Any, Any, int | float]]
+    value_fn: AsyncCallable[[AdGuardHome], int | float]
```

Comment:
These cases are simple. However, what about overloads which return both a sync and async version?

Also consider that `AsyncCallable` isn't something in the stdlib, so new developers won't find any info on it in the Python docs. At best I'd suggest to add an alias for `Coroutine[Any, Any, _T|`, maybe `Coro[_T]` (?), but even that should ideally be added to the stdlib first.

I'd suggest we keep it as is.

Reply:
This does not aim to solve 100% of cases.
If it return a sync and an async version, then it would continue to use the current.

As an example, this one wouldn't change:
https://github.com/home-assistant/core/blob/e4731c07cb96f915621a2c8803a320b2e29e6d5f/homeassistant/config_entries.py#L298

Reply:
Regarding the Python docs, I've now added a docstring, and I think new developers will find it clearer with `AsyncCallable` and thus won't need to look through the Python docs.

Reply:
Regarding the last point about adding `Coro`, I don't see much benefit in it.
A quick scan of the code shows that most references to `Coroutine` are nested inside `Callable`.

I think the third option below is the clearest.
```python
# Before:
value_fn: Callable[[AdGuardHome], Coroutine[Any, Any, int | float]]

# With Coro:
value_fn: Callable[[AdGuardHome], Coro[int | float]]

# With AsyncCallable:
value_fn: AsyncCallable[[AdGuardHome], int | float]
```

Reply:
I'm still -1 one this change. If other people think it's a good idea so be it 🤷🏻

Reply:
I'm with @cdce8p on this one; we have been remove many more of these alias things in the past; mostly to make things more standard Python.